src/app/shared/spinner/spinner.component.ts

```diff
@@ -0,0 +1,39 @@
+import {Component, Input, OnDestroy} from 'angular2/core';
+
+@Component({
+  selector: 'spinner',
+  styleUrls: ['app/shared/spinner/spinner.component.css'],
+  template: `<div [hidden]="!isSpinning" class="spinner"></div>`
+})
+
+export class SpinnerComponent implements OnDestroy {
+
+  private currentTimeout: number;
+  private isSpinning: boolean = false;
+
+  @Input() public delay: number = 300;
+
+  @Input() public set spinning(value: boolean) {
+    if (!value) {
+      this.cancelTimeout();
+      this.isSpinning = false;
+    }
+    if (this.currentTimeout) {
+      return;
+    }
+    this.currentTimeout = setTimeout(() => {
```

Comment:
Looks pretty much equivalent to me.  The rxjs syntax might be slightly cleaner - I'll probably try to use more of it going forward.

FWIW, I'm not sure that delay/timeout stuff will actually be useful for us, so it may end up just getting nuked at some point.  (After we build a couple different pages with spinners on them).